cicecore/cicedynB/general/ice_forcing.F90

```diff
@@ -5398,7 +5409,199 @@ end subroutine get_wave_spec
+         isnw_rhos_out=idx_rhos_max, isnw_Tgrd_out=idx_Tgrd_max, isnw_T_out=idx_T_max)
+
+      if (my_task == master_task) then
+         write (nu_diag,*) ' '
+         write (nu_diag,*) 'Snow aging file:', trim(snw_filename)
+      endif
+
+      if (snw_aging_table == 'snicar') then
+         ! just read the 3d data and pass it in
+
+         call ice_open_nc(snw_filename,fid)
+
+         allocate(snowage_tau  (idx_rhos_max, idx_Tgrd_max, idx_T_max))
+         allocate(snowage_kappa(idx_rhos_max, idx_Tgrd_max, idx_T_max))
+         allocate(snowage_drdt0(idx_rhos_max, idx_Tgrd_max, idx_T_max))
+
+         fieldname = trim(snw_tau_fname)
+         call ice_read_nc(fid,fieldname,snowage_tau,  diag, &
+                          idx_rhos_max,idx_Tgrd_max,idx_T_max)
+         fieldname = trim(snw_kappa_fname)
+         call ice_read_nc(fid,fieldname,snowage_kappa,diag, &
+                          idx_rhos_max,idx_Tgrd_max,idx_T_max)
+         fieldname = trim(snw_drdt0_fname)
+         call ice_read_nc(fid,fieldname,snowage_drdt0,diag, &
+                          idx_rhos_max,idx_Tgrd_max,idx_T_max)
+
+         call ice_close_nc(fid)
+
+         call broadcast_array(snowage_tau  , master_task)
+         call broadcast_array(snowage_kappa, master_task)
+         call broadcast_array(snowage_drdt0, master_task)
+
+         if (my_task == master_task) then
+            write(nu_diag,*) subname,'  '
+            write(nu_diag,*) subname,' Successfully read snow aging properties:'
+            write(nu_diag,*) subname,' snw_aging_table = ',trim(snw_aging_table)
+            write(nu_diag,*) subname,' idx_rhos_max = ',idx_rhos_max
+            write(nu_diag,*) subname,' idx_Tgrd_max = ',idx_Tgrd_max
+            write(nu_diag,*) subname,' idx_T_max    = ',idx_T_max
+            write(nu_diag,*) subname,' Data at rhos, Tgrd, T at first index '
+            write(nu_diag,*) subname,' snoage_tau (1,1,1)         = ',snowage_tau  (1,1,1)
+            write(nu_diag,*) subname,' snoage_kappa (1,1,1)       = ',snowage_kappa(1,1,1)
+            write(nu_diag,*) subname,' snoage_drdt0 (1,1,1)       = ',snowage_drdt0(1,1,1)
+            write(nu_diag,*) subname,' Data at rhos, Tgrd, T at max index'
+            write(nu_diag,*) subname,' snoage_tau (max,max,max)   = ',snowage_tau  (idx_rhos_max, idx_Tgrd_max, idx_T_max)
+            write(nu_diag,*) subname,' snoage_kappa (max,max,max) = ',snowage_kappa(idx_rhos_max, idx_Tgrd_max, idx_T_max)
+            write(nu_diag,*) subname,' snoage_drdt0 (max,max,max) = ',snowage_drdt0(idx_rhos_max, idx_Tgrd_max, idx_T_max)
+         endif
+
+         call icepack_init_parameters(        &
+            snowage_tau_in   = snowage_tau,   &
+            snowage_kappa_in = snowage_kappa, &
+            snowage_drdt0_in = snowage_drdt0 )
+
+         deallocate(snowage_tau)
+         deallocate(snowage_kappa)
+         deallocate(snowage_drdt0)
+
+      else
+         ! read everything and pass it in
```

Comment:
I'm in favor of just reading everything in, if it's all available in the file anyhow.  What is different in the file from the coded version?  I'm wondering if I don't have the most up-to-date file.

Reply:
See the icepack PR.  I set the mode to "file" in the set_nml files because I agree it's the right way to do it.  Also, I have not implemented any table reading in Icepack.  That functionality could be ported there, but I thought Icepack was just going to use the "test" data.

Reply:
Icepack will only use the 'test' data.  If we have both 'snicar' and 'file' in Icepack and remove 'snicar' from CICE, then 'snicar' in Icepack won't get tested.  I'm in favor of removing it completely.  Will wait for N's response.